packages/eas-build-job/src/generic.ts

```diff
@@ -31,5 +31,8 @@ export namespace Generic {
     }),
     expoDevUrl: z.string().url(),
     builderEnvironment: BuilderEnvironmentSchemaZ,
+    // We use this to discern between Android.Job, Ios.Job and Generic.Job.
```

Comment:
We have a `type` field in the `Job` object (I mean `BuildJob`). Maybe we can use this field somehow to distinguish between generic jobs and build jobs?
https://github.com/expo/eas-build/blob/a8c2f0d7b24defc7016987a3d7f400f7808f4a29/packages/eas-build-job/src/android.ts#L69

Reply:
This is a good suggestion – I'm not a huge fan of using `platform` as the discriminator either.

However, I wouldn't want to extend the `Workflow` enum to accommodate generic jobs either (funnily enough, there already is a `Workflow.GENERIC` which means a completely different thing.

We could add a `jobType: "build" | "jobRun"` property, but:
- we'll need to support `jobType: undefined` nonetheless (for backwards compatibility)
- having `platform: undefined` actually has a lot of good consequences, because we use `platform` for many things, so we get to recheck all those for generic jobs.

Given this, what is your thinking?

Reply:
>However, I wouldn't want to extend the Workflow enum to accommodate generic jobs either (funnily enough, there already is a Workflow.GENERIC which means a completely different thing.

Oh, I forgot about it 😅 

>we'll need to support `jobType: undefined` nonetheless (for backwards compatibility)

Maybe we can make `the jobType` required and transform the `undefined` value to `build` by default using the `.default()` option in the Joi/Zod schema?

I mean having a different `jobType` seems like the cleanest approach to me, but it's not a string opinion.

I'm fine with proceeding as it is or adjusting it in a separate PR.

Reply:
I'll proceed with the current state for now, who knows what else we'll do here haha.